packages/sdk/src/types/runtime/groupMask.ts

```diff
@@ -0,0 +1,147 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Context } from '.';
+
+// tslint:disable:no-bitwise
+
+/**
+ * A set of user group IDs. User groups are used to selectively enable several different
+ * properties of actors based on the memberships of the viewing user. All users not assigned
+ * a group are in the `default` group. See [[User.groups]], [[Appearance.enabled]].
+ */
+export class GroupMask extends Set<string> {
+    public static readonly ALL_PACKED = ~0;
+    public static readonly NONE_PACKED = 0;
+
+    // tslint:disable-next-line:variable-name
+    private _allowDefault = true;
+    public get allowDefault() { return this._allowDefault; }
+    public set allowDefault(val) {
+        this._allowDefault = val;
+        this.delete('default');
+    }
+
+    constructor(private context: Context, initialContents: Iterable<string> = null) {
+        super();
+        if (initialContents) {
+            for (const group of initialContents) {
+                if (this.allowDefault || group !== 'default') {
+                    this.getOrAddMapping(group);
+                    super.add(group);
+                }
+            }
+        }
+    }
+
+    public static All(context: Context) {
```

Comment:
Please document all public elements, or mark them @hidden.

Reply:
Okay. Should I also document the super-class methods? I don't know how typedoc handles those.

---
In reply to: [268861446](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/274#discussion_r268861446) [](ancestors = 268861446)